components/service/glean/src/main/java/mozilla/components/service/glean/Glean.kt

```diff
@@ -0,0 +1,103 @@
+
+import mozilla.components.service.glean.config.Configuration
+import mozilla.components.service.glean.net.HttpPingUploader
+import mozilla.components.service.glean.ping.PingMaker
+import mozilla.components.service.glean.storages.ExperimentsStorageEngine
+import mozilla.components.service.glean.storages.StorageEngineManager
+
+object Glean {
+    // Include our singletons of StorageEngineManager and PingMaker
+    private val storageEngineManager = StorageEngineManager()
+    private val pingMaker = PingMaker(storageEngineManager)
+    private val configuration = Configuration()
+    internal var httpPingUploader = HttpPingUploader(configuration)
+
+    private var initialized = false
+    private var metricsEnabled = true
+
+    /**
+     * Initialize glean.
+     */
+    fun initialize() {
+        this.initialized = true
+    }
+
+    /**
+     * Enable or disable metric collection.
+     *
+     * Metric collection is emabled by default.
+     *
+     * When disabled, metrics aren't recorded at all.
+     *
+     * @param enabled When true, enable metric collection.
+     */
+    fun setMetricsEnabled(enabled: Boolean) {
+        this.metricsEnabled = enabled
+    }
+
+    /**
+     * Get the status of metrics enabled.
+     */
+    fun getMetricsEnabled(): Boolean {
+        return this.metricsEnabled
+    }
+
+    /**
+     * Indicate that an experiment is running. Glean will then add an
+     * experiment annotation to the environment which is sent with pings. This
+     * information is not persisted between runs.
+     *
+     * @param experimentId The id of the active experiment (maximum
+     *     30 bytes)
+     * @param branch The experiment branch (maximum 30 bytes)
+     * @param extra Optional metadata to output with the ping
+     */
+    fun setExperimentActive(
+        experimentId: String,
+        branch: String,
+        extra: Map<String, String>? = null
+    ) {
+        ExperimentsStorageEngine.setExperimentActive(experimentId, branch, extra)
+    }
+
+    /**
+     * Indicate that an experiment is no longer running.
+     *
+     * @param experimentId The id of the experiment to deactivate.
+     */
+    fun setExperimentInactive(
+        experimentId: String
+    ) {
+        ExperimentsStorageEngine.setExperimentInactive(experimentId)
+    }
+
+    private fun sendPing(store: String, path: String) {
+        val pingContent = pingMaker.collect(store)
+        this.httpPingUploader.upload(pingContent, path)
+    }
+
+    /**
+     * Handle an event and send the appropriate pings.
+     *
+     * Valid events are:
+     *
+     *   - background: When the application goes to the background.
+     *       This triggers sending of the default ping.
+     *   - default: Event that triggers the default pings.
+     *
+     * @param name The name of the event.
+     */
+    fun handleEvent(name: String) {
+        if (name == "background") {
+            sendPing("baseline", "/baseline")
```

Comment:
I've updated this to store the UUID in `ping_info/ping_id` for now, and updated this PR to WIP until to wait to hear from @mreid-moz on how best to handle that.

Reply:
It's needed in the URL path for skipping duplicate submissions. It's not required inside the ping itself (for example, the [core ping](https://github.com/mozilla-services/mozilla-pipeline-schemas/blob/dev/schemas/telemetry/core/core.9.schema.json) doesn't include it inside the document). If you can't think of a reason why it would be useful on the client-side, I'd lean towards leaving it out.

Reply:
Thanks.  I'll leave it out of the ping content itself then.